test/e2e/common/node/kubelet.go

```diff
@@ -260,3 +262,87 @@ var _ = SIGDescribe("Kubelet with pods in a privileged namespace", func() {
 		})
 	})
 })
+
+var _ = SIGDescribe("Kubelet rejects pods that do not satisfy affinity after restart", func() {
```

Comment:
Can you move the test here actually? https://github.com/kubernetes/kubernetes/blob/a63cc95cbec32e94054921877f08a75e215cc74d/test/e2e_node/restart_test.go

We have examples of restarting kubelet so I don't think you need changes to storage/utils.

Reply:
done, thanks @kannon92